api/blog/blog.service.js

```diff
@@ -29,8 +29,67 @@ class BlogService {
         return ++lastId;
     }
 
-    findAll(cb) {
-        return cb(null, posts)
+    findAll(page, limit, filter, sort, sortDir, cb) {
+        let returnPosts = posts;
+
+        //Filtering
+        if (filter) {
+            returnPosts = returnPosts.filter(post => post.title.startsWith(filter));
+        }
+
+        //Sorting
+        if (sort && sortDir){
+            switch (sort){
+                default:
+                case 'id': {
+                    if (sortDir == 'asc'){
+                        returnPosts = returnPosts.sort((a, b) => { //After the array is with all the elements, we sort by postId alphabetically
+                            return a.id - b.id;
+                        })
+                    } else {
+                        returnPosts = returnPosts.sort((a, b) => { //After the array is with all the elements, we sort by postId alphabetically
+                            return b.id - a.id;
+                        })
+                    }
+                    break;
+                }
+
+                case 'title': {
+                    if (sortDir == 'asc'){
+                        returnPosts = returnPosts.sort((a, b) => { //After the array is with all the elements, we sort by postId alphabetically
+                            return a.title - b.title;
+                        })
+                    } else {
+                        returnPosts = returnPosts.sort((a, b) => { //After the array is with all the elements, we sort by postId alphabetically
+                            return b.title - a.title;
+                        })
+                    }
+                    break
+                }
+
+                case 'author': {
+
+                    if (sortDir == 'asc'){
+                        returnPosts = returnPosts.sort((a, b) => { //After the array is with all the elements, we sort by postId alphabetically
+                            return a.author - b.author;
+                        })
+                    } else {
+                        returnPosts = returnPosts.sort((a, b) => { //After the array is with all the elements, we sort by postId alphabetically
+                            return b.author - a.author;
+                        })
+                    }
+                    break;
+                }
+            }
+        }
```

Comment:
backend will change using mongoose and we won't need anymore the switch, because databases has this mechanism built in, but let leave it this way